src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepository.groovy

```diff
@@ -90,26 +99,36 @@ class GitRepository implements ScmRepository {
     }
 
     @Override
+    void dropTag(String tagName) {
+        repository.tag.remove(names: [tagName])
+    }
+
+    @Override
     void push(ScmIdentity identity, ScmPushOptions pushOptions) {
         push(identity, pushOptions, false)
     }
 
     void push(ScmIdentity identity, ScmPushOptions pushOptions, boolean all) {
-        identity.useDefault ? callPush(pushOptions, all) : callLowLevelPush(identity, pushOptions, all)
+        callPush(identity, pushOptions, all)
     }
 
-    private void callPush(ScmPushOptions pushOptions, boolean all) {
+    private void callPush(ScmIdentity identity, ScmPushOptions pushOptions, boolean all) {
         if (!pushOptions.pushTagsOnly) {
-            repository.push(remote: pushOptions.remote, all: all)
+            def regularPush = pushCommand(identity, pushOptions.remote, all)
```

Comment:
the only difference between `pushCommand` here and next invocation is `setPushTags`, so how about:

```
def pushCommand = pushCommand(...)
if(pushOptions.pushTagsOnly) {
    pushCommand = pushCommand.setPushTags()
}
```

Also please use types instead of `def`, as i do in other places

Reply:
Since this comment is also true for situation that been before, I figured you had some reason to leave it that way. Happy to change it though.